### DISCUSSION TITLE
Get NodeJS spawn to show ddev import-db progress

### DISCUSSION THREAD

**Discussion by Mathijs-Zuiderlicht** [ddev/ddev#6483] [Q&A] [answered]
Hi!

I'm tinkering with nodejs and trying to get node to execute some ddev commands.
My goal is to write the progress of a db import to the console.

If I run _DDEV import-db --file=somefile.sql_ then I get a nice progress bar that shows me how long the import may take.
However, the same (?) command when executed in node does not, and only logs "Successfully imported database 'db' for myDB"

Here's a snippet of the code. In all 3 console.logs I get nothing but the final message.

```
                const importCommand = 'wsl -d Ubuntu --cd ' + appRoot + ` ddev import-db --file=${dumpFileName}`;
                const importProcess = spawn(importCommand, { shell: true });

                importProcess.stdout.on('data', (data) => {
                    console.log(`Output: ${data}`);
                });

                importProcess.on('message', (message) => {
                    console.log('Message:', message);
                });

                importProcess.stderr.on('data', (data) => {
                    console.error(`Error output: ${data}`);
                });
```

I have looked at the DDEV source and it seems to use pv (Pipe Viewer) to show the progress. 
https://github.com/ddev/ddev/blob/4e22fe8284e5ccfb129eaf6522a45724be54ddea/pkg/ddevapp/ddevapp.go#L746

I would expect that to return something to stdout. But allas.
Can it be done? Any ideas?

**Comment by stasadev** [comment, marked as answer]
Hi @Mathijs-Zuiderlicht,

Please test:

- https://github.com/ddev/ddev/pull/6484#issuecomment-2288692836

**Comment by Mathijs-Zuiderlicht** [comment]
This works:

`const importCommand = 'wsl -d Ubuntu pv -f ' + dumpFilePath + ' | wsl -d Ubuntu --cd ' + appRoot + ' ddev mysql -uroot -proot ' + myDB`
